llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -40,6 +40,7 @@
 #include <tuple>
 #include <type_traits>
 #include <utility>
+#include <variant>
```

Comment:
I don't think we want to pull in variant in this header;  I'm worried about the  effect on compilation times and some implementations not being optimized very well. Can you implement it without variant?